internal/sync/l1Collector.go

```diff
@@ -566,6 +567,35 @@ func (l *l1Collector) fetchPendingBlock() {
 	l.pendingBlock = pendingBlock
 }
 
+// fetchContractCode fetch the code of the contract from the Feeder Gateway.
+func (l *l1Collector) fetchContractCode(stateDiff *types2.StateDiff) *CollectorDiff {
+	collectedDiff := &CollectorDiff{
+		stateDiff: stateDiff,
+	}
+	for _, deployedContract := range stateDiff.DeployedContracts {
+
+		contractFromApi, err := l.client.GetFullContractRaw(deployedContract.Address.Hex0x(), "",
+			strconv.FormatInt(stateDiff.BlockNumber, 10))
+		if err != nil {
+			l.logger.With("Block Number", stateDiff.BlockNumber,
+				"Contract Address", deployedContract.Address.Hex0x()).
+				Error("Error getting full contract")
+			return collectedDiff
+		}
+
+		contract := new(types2.Contract)
+		err = contract.UnmarshalRaw(contractFromApi)
+		if err != nil {
+			l.logger.With("Block Number", stateDiff.BlockNumber,
+				"Contract Address", deployedContract.Address.Hex0x()).
+				Error("Error unmarshalling contract")
+			return collectedDiff
+		}
+		collectedDiff.Code[deployedContract.Address.Hex0x()] = contract
+	}
+	return collectedDiff
+}
+
```

Comment:
These lines of code are identical to the ones in the API collector. Please find a way to remove this duplication.

Reply:
done, put on the utils package